pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -0,0 +1,678 @@
+				StringResponseDummy{pam.TextInfo, "answer to TextInfo"},
+			},
+		},
+
+		"binary requests": {
+			requests: []pam.ConvRequest{
+				NewBinaryRequestDummy(nil),
+				NewBinaryRequestDummy(pam.BinaryPointer(&[]byte{})),
+				NewBinaryRequestDummy(pam.BinaryPointer(&[]byte{0xFF, 0x00, 0xBA, 0xAB})),
+				NewBinaryRequestDummy(pam.BinaryPointer(&[]byte{0x55})),
+			},
+			expectedResponses: []pam.ConvResponse{
+				&BinaryResponseDummy{pam.BinaryPointer(&[]byte{})},
+				&BinaryResponseDummy{nil},
+				&BinaryResponseDummy{pam.BinaryPointer(&[]byte{0x53})},
+				&BinaryResponseDummy{pam.BinaryPointer(&[]byte{0xAF, 0x00, 0xBA, 0xAC})},
+			},
+		},
+
+		"mixed requests": {
+			requests: []pam.ConvRequest{
+				NewBinaryRequestDummy(nil),
+				pam.NewStringConvRequest(pam.PromptEchoOff, "PromptEchoOff"),
+				NewBinaryRequestDummy(pam.BinaryPointer(&[]byte{})),
+				pam.NewStringConvRequest(pam.PromptEchoOn, "PromptEchoOn"),
+				NewBinaryRequestDummy(pam.BinaryPointer(&[]byte{0xFF, 0x00, 0xBA, 0xAB})),
+				pam.NewStringConvRequest(pam.ErrorMsg, "ErrorMsg"),
+				NewBinaryRequestDummy(pam.BinaryPointer(&[]byte{0x55})),
+				pam.NewStringConvRequest(pam.TextInfo, "TextInfo"),
+			},
+			expectedResponses: []pam.ConvResponse{
+				&BinaryResponseDummy{pam.BinaryPointer(&[]byte{})},
+				StringResponseDummy{pam.PromptEchoOff, "PromptEchoOff"},
+				&BinaryResponseDummy{pam.BinaryPointer(&[]byte{0x55})},
+				StringResponseDummy{pam.PromptEchoOn, "PromptEchoOn"},
+				&BinaryResponseDummy{nil},
+				StringResponseDummy{pam.ErrorMsg, "ErrorMsg"},
+				&BinaryResponseDummy{pam.BinaryPointer(&[]byte{0xAF, 0x00, 0xBA, 0xAC})},
+				StringResponseDummy{pam.TextInfo, "TextInfo"},
+			},
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			require.Equalf(t, len(tc.expectedResponses), len(tc.requests),
+				"Setup: mismatch on expectations / requests numbers")
+
+			convHandler := &multiConvHandler{
+				t:                t,
+				expectedRequests: tc.requests,
+				responses:        tc.expectedResponses,
+			}
+			tx := NewModuleTransactionDummy(convHandler)
+
+			responses, err := tx.StartConvMulti(tc.requests)
+			require.ErrorIs(t, err, tc.expectedError)
+			require.Equal(t, len(tc.requests), convHandler.timesCalled)
+
+			if tc.expectedError != nil {
+				require.Nil(t, responses)
+				return
+			}
+
+			require.NotNil(t, responses)
+			require.Len(t, responses, len(tc.requests))
+
+			for i, res := range responses {
+				expected := tc.expectedResponses[i]
+				require.Equal(t, expected.Style(), res.Style())
+
+				switch r := res.(type) {
+				case pam.BinaryConvResponse:
+					expBinRes, ok := expected.(pam.BinaryConvResponse)
+					require.True(t, ok)
+					eb, err := expBinRes.Decode(bytesPointerDecoder)
+					require.NoError(t, err)
+					bytes, err := r.Decode(bytesPointerDecoder)
+					require.NoError(t, err)
+					require.Equal(t, eb, bytes)
+				default:
```

Comment:
Even though this already points out to `pam.StringConvResponse`, I would use it explicitly (i.e. instead of ```default:```, use ```case pam.StringConvResponse:``` to indicate better what this require is about.

Reply:
Indeed, betther, that's handled too.